### PR TITLE
Consultation should have an option to mark bed number back end

### PR DESCRIPTION
**Pull request by sanoop-pk**

### CODE REVIEW

File: care/facility/migrations/0094_auto_20200416_1414.py [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+    operations = [
+        migrations.AddField(
+            model_name='patientconsultation',
+            name='bed_number',
+            field=models.CharField(blank=True, max_length=100, null=True),
+        ),
+        migrations.AlterField(
+            model_name='facility',
+            name='facility_type',
+            field=models.IntegerField(choices=[(1, 'Educational Inst'), (2, 'Private Hospital'), (3, 'Other'), (4, 'Hostel'), (5, 'Hotel'), (6, 'Lodge'), (7, 'TeleMedicine'), (8, 'Govt Hospital'), (9, 'Labs'), (800, 'Primary Health Centres'), (801, '24x7 Public Health Centres'), (802, 'Family Health Centres'), (803, 'Community Health Centres'), (820, 'Urban Primary Health Center'), (830, 'Taluk Hospitals'), (831, 'Taluk Headquarters Hospitals'), (840, 'Women and Child Health Centres'), (850, 'General hospitals'), (860, 'District Hospitals'), (870, 'Govt Medical College Hospitals'), (950, 'Corona Testing Labs')]),
```
Comment (vigneshhari) [inline care/facility/migrations/0094_auto_20200416_1414.py:21]: Codacy Issue found: [Line too long (620/120)](https://app.codacy.com/gh/coronasafe/care/pullRequest?prid=5368960)

File: care/facility/api/viewsets/patient_consultation.py [hunk 2]
```diff
@@ -40,6 +40,14 @@ def list(self, request, *args, **kwargs):
         """
         return super().list(request, *args, **kwargs)
 
+    def perform_create(self, serializer):
+        try:
+            if not self.request.data["admitted"]:
+                self.request.data["bed_number"] = ""
+        except KeyError:
+            self.request.data["bed_number"] = ""
+        serializer.save()
```
Comment (anroopak) [inline care/facility/api/viewsets/patient_consultation.py:49]: `perform_create` is called after validation using serializer. Also serializer will be handle a good amount of security things like SQL injection, etc.  Hence we should not be touching the`self.request.data` and only be using `serializer.validated_data`. 

Also conventionally such validation / data cleaning should be part of the `.validation()` of the serializer. Please move this function to the serializer.

File: care/facility/migrations/0094_auto_20200416_1414.py (hunk 1 above)
Comment (vigneshhari) [inline care/facility/migrations/0094_auto_20200416_1414.py:21]: Codacy Issue found: [Line too long (620/120)](https://app.codacy.com/gh/coronasafe/care/pullRequest?prid=5368960)

Comment (vigneshhari) [inline care/facility/migrations/0094_auto_20200416_1414.py:21]: Codacy Issue found: [Line too long (620/120)](https://app.codacy.com/gh/coronasafe/care/pullRequest?prid=5368960)

**Pull request merged**